lib/ContextModule.js

```diff
@@ -140,6 +140,28 @@ class ContextModule extends Module {
 					this.addBlock(block);
 				}
 
+			} else if(this.async === "weak-context") {
+
+				// store the dependences in a different key than `this.dependences`
+				// to prevent them from being bundled in the parent
+				this.weakDependencies = dependencies;
```

Comment:
Instead of doing this, you need to set a `weak` property to `true` on each of `this.dependencies`.